source/extensions/retry/priority/previous_priorities/previous_priorities.h

```diff
@@ -13,15 +13,16 @@ class PreviousPrioritiesRetryPriority : public Upstream::RetryPriority {
 public:
   PreviousPrioritiesRetryPriority(uint32_t update_frequency, uint32_t max_retries)
       : update_frequency_(update_frequency) {
-    attempted_priorities_.reserve(max_retries);
+    attempted_hosts_.reserve(max_retries);
   }
 
   const Upstream::HealthyAndDegradedLoad&
   determinePriorityLoad(const Upstream::PrioritySet& priority_set,
-                        const Upstream::HealthyAndDegradedLoad& original_priority_load) override;
+                        const Upstream::HealthyAndDegradedLoad& original_priority_load,
+                        const PriorityMappingFunc& priority_mapping_func) override;
 
   void onHostAttempted(Upstream::HostDescriptionConstSharedPtr attempted_host) override {
-    attempted_priorities_.emplace_back(attempted_host->priority());
+    attempted_hosts_.emplace_back(attempted_host);
```

Comment:
guessing the change to do hosts instead of just the priorities here was in order to retain the update frequency semantics even when we fail to determine the priority?

Reply:
No, it was because I didn't have the mapping cb function in this context, and I thought it was more leaky to store the cluster name and priority, because what if a different mapping function at some point needed another field from the host?